### DISCUSSION TITLE
How to import Fenix to Obtainium from an HTML source? Tried but in vain.

### DISCUSSION THREAD

**Discussion by PumpkinJui** [ImranR98/Obtainium#2024] [Q&A] [answered]
Hi everyone. I want to import Fenix (aka Firefox for Android) to Obtainium in HTML mode, but despite hours' effort, I got nothing except `error: Could not find a suitable release`. I wonder if anyone could help me out.

First of all, I should explain why I need to use HTML mode. As I live in Mainland China, Google Play is blocked and other sources are either blocked or extremely slow. Firefox in the store is a specially provided version and not the same as the international one. So I have to get my apk from `https://ftp.mozilla.org/pub/`, known as Mozilla FTP.

The directory structure is neat. For example, the 133.0 version, arm64-v8a architecture apk of Fenix is located at `https://ftp.mozilla.org/pub/fenix/releases/133.0/android/fenix-133.0-android-arm64-v8a/fenix-133.0.multi.android-arm64-v8a.apk`. `fenix/releases/` holds all releases (compared with candidates and nightly), including stable releases and beta releases, using version name as the directory name. Beta uses another package name and has a trailing `b` in the directory name, and I don't want to use it at all. Then `android/` holds all architectures' apk of this version.

So, I used `https://ftp.mozilla.org/pub/fenix/releases/` as the source link, and `\d{2,3}\.\d/android/fenix-\d{2,3}\.\d-android-arm64-v8a/.+\.apk` as the custom apk link filter. The regex is exactly correct, and everyone can check it with [regexlearn playground](https://regexlearn.com/playground); but Obtainium just responded with `error: Could not find a suitable release`. I also tried using some other variants, all of which correctly matched the link but just responded as error in Obtainium. I even split it into different parts and add them as intermediate links, but failed as well.

I am at the latest version (v1.1.34) and I've read [the wiki](https://wiki.obtainium.imranr.dev/sources/) but found nothing useful to solve this. Could anyone explain where I was wrong, how to correct it, or even whether Obtainium is capable of doing that? Thank you all for any help.

**Comment by DarkCrypt** [comment, marked as answer]
A working config for Firefox is on Obtainium's config website.

https://apps.obtainium.imranr.dev/?categories=browser

**Reply by PumpkinJui** [reply]
Oh, I didn't see it. Thank you so much for reminding me!
